orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -324,6 +324,7 @@ class MonitoredDeployStrategy implements Strategy {
     }
 
     if (stageData.deploymentMonitor?.id) {
+      internalStageData.failure = false;
```

Comment:
this doens't really do anything since the `internalStageData` isn't passed to the stage below
you would need to add it to the `evalContext` on line 332. I would suggest change L332 to:
```java
evalContext + [hasDeploymentFailed: false]
```